packages/web3-common/src/web3_base_wallet.ts

```diff
@@ -0,0 +1,25 @@
+export interface Web3BaseWalletAccount {
+	[key: string]: unknown;
+	readonly address: string;
+	readonly privateKey: string;
+	readonly signTransaction: (tx: Record<string, unknown>) => string;
+	readonly sign: (data: Record<string, unknown> | string) => string;
+	readonly encrypt: (password: string, options?: Record<string, unknown>) => string;
+}
+export type Web3EncryptedWallet = string;
+
+export interface Web3BaseWallet<T = Web3BaseWalletAccount> {
+	create: (numberOfAccounts: number, entropy: string) => Web3BaseWallet<T>;
+	add: (account: T) => boolean;
+	get: (addressOrIndex: string | number) => T;
+	remove: (addressOrIndex: string | number) => boolean;
+	clear: () => Web3BaseWallet<T>;
+	encrypt: (password: string, options?: Record<string, unknown>) => T[];
+	decrypt: (
+		encryptedWallet: Web3EncryptedWallet[],
+		password: string,
+		options?: Record<string, unknown>,
+	) => Web3BaseWallet<T>;
+	save: (password: string, keyName?: string) => boolean | never;
+	load: (password: string, keyName?: string) => Web3BaseWallet<T>;
```

Comment:
`keyName` would be the name of the file? In which case I would suggest renaming to `fileName` to be explicit

Reply:
Its the default behaviour of the current implementation, that parameter is optional. And each wallet implementation can have some default `keyName | fileName` to use internally.